components/contexts/Web3Context.js

```diff
@@ -41,46 +37,38 @@ export const Web3Provider = ({ children }) => {
   }, [])
 
   useEffect(() => {
-    if (networkId && networkId !== NetworkId) {
-      setError(`Please switch to ${NetworkName}`)
+    if (networkId && networkId !== Network.id) {
+      setNetworkError(
+        `Please switch to ${ethers.providers.getNetwork(Network.id).name}`,
+      )
+    } else {
+      setNetworkError(false)
     }
   }, [networkId])
 
   const connect = useCallback(
     async function connect() {
-      const providerInstance = await web3Modal.connect()
-      const web3Instance = new Web3(providerInstance)
-
-      // Set provider:
-      setProvider(providerInstance)
-      setWeb3(web3Instance)
-
-      // Set accounts:
-      setAccounts(await web3Instance.eth.getAccounts())
-
-      // Set network:
-      setNetworkId(web3Instance.currentProvider.chainId)
+      const _web3 = await web3Modal.connect()
+      const _provider = new ethers.providers.Web3Provider(_web3, 'any')
+      const _signer = _provider.getSigner()
+      const _address = await _signer.getAddress()
+      const _network = await _provider.getNetwork()
+
+      setProvider(_provider)
+      setWeb3(_web3)
+      setSigner(_signer)
+      setNetworkId(_network.chainId)
+      setAddress(_address)
```

Comment:
Major differences I've seen with ethers.js (vs. web3.js) so far:

* The ethers provider inherits from the wallet selected with web3wallet, but we can also create multiple providers for other uses (ie. a provider using alchemy for scalable/stable read-only operations)
* Instead of an `accounts` array, ethers.js gets a `signer` object from the provider, which comes with utility, including the obviously helpful `getAddress` method ;)
* Contract initialization in ethers seems to take a signer, which loads that key into the instance making tx signing way easier
* Ethers has it's own events that seem to inject more helpful information than web3.js, and can also bind directly to contract events/logs in a very clean interface

Reply:
From my limited experience, ethers seems to have the easier usage and if you don’t need some obscure thing that maybe only web3 does or can be integrated with, is probably the better choice in most use cases. 👍

Reply:
Awesome! Also it's built into hardhat tests, so all the stuff we're familiar with on testing is available in the dApp (like easy contract <> signer connecting with `contract.connect(signerB)`)